Java/bit_manipulation/swap.java

```diff
@@ -0,0 +1,50 @@
+//Program to swap two numbers using bit manipulation.
+/*
+XOR operation can be defined as 
+a	b	a^b
+0	0	0
+0	1	1
+1	0	1
+1	1	1
+
+
+
+*/
+import java.util.Scanner;
+public class swap {
+
+	public static void main(String[] args) {
+		Scanner sc=new Scanner(System.in);
+		System.out.println("Enter No a=");
+		int a=sc.nextInt();
+		System.out.println("Enter No a=");
+		int b=sc.nextInt();
+		//swapping 
+		a=a^b;

```

Comment:
Move the logic part to a new function, the main function should only have the driver code.